### ISSUE TITLE
Integrate with .NET Core release pipeline

### ISSUE THREAD

**Issue by mthalman** [dotnet/dotnet-docker#1267]
It would be nice to have an integrated solution for building/testing/publishing the .NET Core Docker images as part of the .NET Core product's release pipeline.  Today, it's completely separate with just a manual task placeholder in the current release pipeline representing the task to publish the images.  It would be slick to have that task integrate with our current pipeline instead.

**Comment by MichaelSimons** [comment]
Capturing some thoughts on how this could be implemented.

Release Pipeline (Internal): https://dev.azure.com/dnceng/internal/_release?definitionId=5&view=mine&_a=releases

Required Dependencies:
1. [MCR flow, programmatic ingestion confirmation](https://github.com/microsoft/mcr/issues/635)
1. [Git commit based Dockerfile links](https://github.com/dotnet/dotnet-docker/issues/1436)

Manual Preconditions:
1. .NET Core Docker release PR 
    1. Prepared by .NET Core Docker devs
    1. Includes all Dockerfile, manifest, readme, etc. changes for the release
    1. Should be reviewed
    1. Green PR Validation

Release Pipeline Integration:
1. Prepare Docker Images
    1. Dependency:  Publish of .NET Core assets to blob storage (Wait For Publish To Complete?)
    1. Merge .NET Core Docker release PR
        1. Input: PR #
        1. Note:  PR must be merged prior to building in order to ensure the correct commit shas are included in the readmes
    1. Trigger Build 
        1. Queue Build stage
            1. Input: Release version (get transformed into a build filter to only build released versions)
        1. Store BuildId
        1. Note:  This may require one stage to queue the build and another to wait for completion
1. Validate and publish Docker Images
    1. Dependency: NuGet publish (NuGet Push and Validate)
    1. Trigger Build
        1. Queue Test and Publish stages
            1. Input: Release version (get transformed into a build filter to only build released versions)
            1. Input: Build stage build id
        1. Wait on MCR ingestion completion (MCR ingestion confirmation)
        1. Note:  This may require one stage to queue the build and another to wait for completion

**Comment by lbussell** [comment]
Superseded by https://github.com/dotnet/dotnet-docker-internal/issues/7431
